cmd/root.go

```diff
@@ -42,6 +48,73 @@ var rootCmd = &cobra.Command{
 		}
 
 		p := tea.NewProgram(tui.New(root), tea.WithAltScreen(), tea.WithMouseAllMotion())
+
+		if watch {
+			watcher, err := fsnotify.NewWatcher()
+			if err != nil {
+				p.Send(tui.UpdateSlidesMsg{NewRoot: createErrorSlide(err, "none")})
+				return nil
+			}
+			defer watcher.Close()
+
+			absPath, err := filepath.Abs(filename)
+			if err != nil {
+				p.Send(tui.UpdateSlidesMsg{NewRoot: createErrorSlide(err, "none")})
+				return nil
+			}
+
+			if err := watcher.Add(filepath.Dir(absPath)); err != nil {
+				p.Send(tui.UpdateSlidesMsg{NewRoot: createErrorSlide(err, "none")})
+				return nil
+			}
+
+			go func() {
+				var debounceTimer *time.Timer
+
+				for {
+					select {
+					case event, ok := <-watcher.Events:
+						if !ok {
+							return
+						}
+
+						if event.Name == absPath || event.Name == filename ||
+							strings.HasSuffix(event.Name, "~") ||
+							strings.HasPrefix(event.Name, absPath+".") {
+							if event.Has(fsnotify.Write) || event.Has(fsnotify.Create) {
+								if debounceTimer != nil {
+									debounceTimer.Stop()
+								}
+								debounceTimer = time.NewTimer(100 * time.Millisecond)
+
+								go func() {
+									<-debounceTimer.C
```

Comment:
A new goroutine is started for each change event, and it waits on `debounceTimer.C` to execute. This approach can lead to a goroutine leak, because calling `.Stop()` on a `time.Timer` prevents the timer from firing, which means its channel will block indefinitely. As a result, any goroutines waiting on that channel won't terminate. A better alternative would be to use `time.AfterFunc`, which avoids this issue and simplifies the debounce logic.

```suggestion
debounceTimer = time.AfterFunc(100 * time.Millisecond, func() {
	data, err := os.ReadFile(filename)
	...
})
```

Reply:
Resolved in ac66dbf1